### PR TITLE
Auto add close button to modal dialogs

### PR DESCRIPTION
**Pull request by ghost**
Related:
- https://github.com/wet-boew/GCWeb/issues/1156
- https://github.com/wet-boew/wet-boew/pull/7909

Also AJAX created Modals don't get button created.

### CODE REVIEW

**Comment by shawnthompson** [comment]
Travis build is fail, needs at PhantomJS test. Need to figure out why.

cc/ @duboisp @LaurentGoderre @kodecount

**Comment by kodecount** [comment]
See a demo: https://kodecount.ca/demos/overlays/

**Comment by kodecount** [comment]
As for the AJAX this maybe out of our control as it being created from another page (DOM). So in this situation the most we could do I think is recommend them to add one unless their another area we call the code.

**Comment by LaurentGoderre** [comment]
It's possible that the build is failing because this PR introduces a JS error.

**Comment by kodecount** [comment]
Could have something to do with mixing pure JavaScript and jQuery if the build is fussy about that.

File: src/plugins/lightbox/lightbox.js [hunk 1]
```diff
@@ -142,10 +143,42 @@ var componentName = "wb-lbx",
...
 						$content = this.contentContainer,
 						$wrap = this.wrap,
+						$modal = $wrap.find( ".modal-dialog" ),
 						$buttons = $wrap.find( ".mfp-close, .mfp-arrow" ),
 						len = $buttons.length,
 						i, button;
 
+					if ( $modal && $modal.length !== 0 ) {
+						var footer = $modal.find( ".modal-footer" )[ 0 ],
+							hasFooter = ( footer && footer.length !== 0 ) ? true : false,
```
Comment (kodecount) [inline src/plugins/lightbox/lightbox.js]: With `( footer && footer.length !== 0 )` maybe it should be `( footer !== null )`.

Comment (LaurentGoderre) [inline src/plugins/lightbox/lightbox.js]: Yeah and that line is over-complicated, you can do just:

```js
var $footer = $modal.find( ".modal-footer" )[ 0 ],
hasFooter = footer !== null
```

Comment (kodecount) [inline src/plugins/lightbox/lightbox.js]: Okay. When I get a chance I'll fix that line.

**Comment by kodecount** [comment]
Figured out how to add it to AJAX content.

**Comment by LaurentGoderre** [comment]
The build is still failing

**Comment by kodecount** [comment]
I'm gonna be offline for a little while (could be a week or more), but when I get back I can work on this.

**Comment by kodecount** [comment]
If this fails I'm at a lost. The code was broken, so should work now.

**Comment by shawnthompson** [comment]
https://shawnthompson.github.io/kodecount-modal-close/unmin/demos/overlay/overlay-en.html

**Comment by lucas-hay** [comment]
@kodecount is there a way you could code this so the close button could be configured by the user. (allow user to add class name, customized text, etc)

**Comment by kodecount** [comment]
If we use `data-wb-lbx` we have to tell it to ignore the settings for button. Easier solution is to use `data-wb-lbx-btn` and keep the current settings as the default if nothing is set.

File: src/plugins/lightbox/lightbox.js [hunk 2]
```diff
@@ -253,6 +258,38 @@ var componentName = "wb-lbx",
...
+					spanTextFtr = footer.innerHTML + i18nText.tClose;
+				} else {
+					footer = document.createElement( "div" );
+					footer.setAttribute( "class", "modal-footer" );
+					footer.style.background = "#fff";
+					spanTextFtr = i18nText.tClose;
+				}
+				spanTextFtr = spanTextFtr.replace( "'", "&#39;" );
+
+				overlayCloseFtr = "<button type='button' id='ftrClose' class='btn btn-sm btn-default pull-left " + closeClassFtr +
```
Comment (duboisp) [inline src/plugins/lightbox/lightbox.js]: Can you change ```btn-default``` for ```btn-primary``` and remove the ```id='ftrClose'```

File: src/plugins/overlay/overlay-en.hbs [hunk 3]
```diff
@@ -356,7 +355,6 @@ wb.doc.on( "click vclick", "#overlay-open-btn", function( event ) {
 				</ol>
 			</li>
 		</ol>
-		<button class="btn btn-primary popup-modal-dismiss" type="button">Close modal popup</button>
```
Comment (duboisp) [inline src/plugins/overlay/overlay-en.hbs]: Can you add a blurb in the overlay/lighbox documentation on how a author can add his own custom button. May be just reuse this working example and specify that it's need to be in a modal-footer.

**Review by duboisp — CHANGES_REQUESTED** [review]

**Comment by kodecount** [comment]
Do we require the ability for the user to customize the button or can we simply explain that by adding their own button to the modal footer well replace the default button? Also should we have the ability to turn this off? In the event we turned this off it would re-enable accessibility issues again.

**Comment by kodecount** [comment]
Also on another note if I'm to change the default visual appeal from `btn-default` to `btn-primary` should this also take affect on the PR that was merged for just overlays?

**Comment by duboisp** [comment]
The following is my opinion, may others have different thought.

I would say to add a new section in the documentation page like "about the close button" and "simply explain that by adding their own button to the modal footer well replace the default button" followed by a code same on how to do it.

I do not think we should allow the ability to turn this off as it is to solve an accessibility issue. If you want you can quickly mention why in the docs under "about the close button".

> Also on another note if I'm to change the default visual appeal from btn-default to btn-primary should this also take affect on the PR that was merged for just overlays?

I would said yes, and that could be a separate PR

**Comment by kodecount** [comment]
Thanks for the input I agree. Not sure if @lucas-hay has any arguments?

**Comment by kodecount** [comment]
I'll produce with add an example if he has none.

**Comment by lucas-hay** [comment]
@kodecount I agree. Thanks for the idea @duboisp

**Comment by kodecount** [comment]
All updated.

**Review by duboisp — APPROVED** [review]

**Pull request merged**